crates/extensions/tedge_log_manager/src/lib.rs

```diff
@@ -230,6 +247,25 @@ impl IntoIterator for &LogManagerBuilder {
             MappingSender::new(self.box_builder.get_sender(), |cmd: GenericCommandState| {
                 LogUploadCmd::try_from(cmd).map(LogInput::LogUploadCmd).ok()
             });
-        vec![(OperationType::LogUpload.to_string(), sender.into())].into_iter()
+        let metadata_sync_sender =
+            MappingSender::new(self.box_builder.get_sender(), |cmd: GenericCommandState| {
+                if cmd.is_finished() {
+                    Some(LogInput::LogMetadataSync(()))
+                } else {
+                    None
+                }
+            });
+        vec![
+            (OperationType::LogUpload.to_string(), sender.into()),
+            (
+                OperationType::SoftwareUpdate.to_string(),
+                metadata_sync_sender.sender_clone(),
+            ),
+            (
+                OperationType::ConfigUpdate.to_string(),
```

Comment:
This line is the causing the `Config update request not processed when operation is disabled for tedge-agent` to fail, as this makes the `WorkflowActor` "think" that this `LogManagerActor` is a built-in operation handler for `config_update` operations rather than treating it just as an "observer" of `config_update` state transitions. So, when it receives the `init` message, it changes it to `scheduled` and `executing` states and sends it to the `LogManagerActor` which just ignores those messages as it really can't "handle" `config_update` operations.

To fix this, we need to bring in this distinction between an operation handler and a casual observer so that `LogManagerActor` can register just as the observer and not the handler.